### PR TITLE
Bluetooth: Controller: llcp: CTE request: Add handling and tests for LL_UNKNOWN_RSP 

### LINKED ISSUE #45708: Bluetooth: Controller: llcp: CTE request control procedure has missing support for LL_UNKNOWN_RSP
**Describe the bug**
The CTE request control procedure has missing implementation of LL_UNKNOWN_RSP.
That causes the Controller to fail two conformance tests: `LL/CON/CEN/BV-59-C` and `LL/CON/PER/BV-63-C`.
It fails due to not generate HCI_LE_CTE_Request_Failed event.

**To Reproduce**
1. Build `hci_uart` sample.
2. Run conformance tests (special hardware and software is required) and see results.
3. Tests: `LL/CON/CEN/BV-59-C` and `LL/CON/PER/BV-63-C` does not pass.

**Expected behavior**
The Controller should pass the tests.

**Impact**
Blocking possibility to obtain QDID for controller with Direction Finding functionality.

### PR DESCRIPTION
**Pull request by ppryga-nordic**
There was missing handling of LL_UNKNOWN_RSP in CTE request control
procedure. In case there is a pending CTE request and peer responses
with LL_UNKNOWN_RSP then Host should be notified with HCI_LE_CTE_-
Request_Failed event. The pending CTE request procedure should be
completed.

The CTE request procedure should be enabled only if a peer supports
CTE response feature. That information can be obtained by feature
exchange procedure. If there were no feature exchange then CTE
request feature may be disabled if a peer responses with LL_UNKNOWN_RSP
for a CTE request.

The implementation of ll_df_set_conn_cte_req_enable was checking if
CTE response feature is supported only when there was feature exchange.
There was missing possibility to stop CTE request if a peer responded
with LL_UNKNOWN_RSP for an earlier CTE request.

The commit changes the implementation of ll_df_set_conn_cte_req_enable.
The CTE response feature check is moved to ull_cp_cte_req function,
because it belongs more to control procedure than to function that
handles Host request to start the procedure.
Second change is related with use of conn->llcp.fex.features_used.
It stores information about features supported by peer. It does
not depend on execution of the feature exchange control procedure.

By the way, there were removed else statement in ll_df_set_conn_cte_-
req_enable because it was not needed.

Add test that verify correct behavior of the CTE request procedure
in case of reception of LL_UNKNOWN_RSP PDU from peer device.

Fixes not working CTE request control procedure unit tests.
Adds small refactoring related changes to CTE request control procedure unit tests.

There was missing code responsible for handling of unexpected response
for CTE request. The commit adds code that will terminate connection
in case a peer device responses with unexpected control PDU that is not
a remote procedure request.

Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/45708.

### CODE REVIEW

**Comment by ppryga** [comment]
Added commit that removes a test that was no longer valid one. More information in commit message.

**Comment by ppryga** [comment]
@thoh-ot @cvinayak @kruithofa @erbr-ot could you take a look on the PR?

**Review by thoh-ot — CHANGES_REQUESTED** [review]

File: tests/bluetooth/controller/ctrl_cte_req/src/main.c [hunk 1]
```diff
@@ -722,6 +722,97 @@ void test_cte_req_reject_inv_ll_param_peripheral_remote(void)
...
+	lt_rx_q_is_empty(&conn);
+
+	/* Rx */
+	lt_tx(LL_UNKNOWN_RSP, &conn, &unknown_rsp);
+
+	/* Done */
+	event_done(&conn);
+
+	/* Receive notification of sampled CTE response */
+	ut_rx_pdu(LL_UNKNOWN_RSP, &ntf, &remote_reject_ext_ind);
```
[the remark this answers is not in the document]
Comment (ppryga) [reply tests/bluetooth/controller/ctrl_cte_req/src/main.c]: Thank you @thoh-ot! Good catch. It should be an address to `unknown_rsp` instance.

**Review by kruithofa — DISMISSED** [review]

**Review by thoh-ot — DISMISSED** [review]

**Comment by ppryga** [comment]
I have to rebase and fix merge conflicts.
Also I've spot that there were issues with unit tests, hence I've added 3 new commits that address:
- wrong value assigned to `CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`
- missing CTE response feature in features supported by peer device
- missing connection handle assignment to global connection handle variable

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_llcp.c [hunk 2]
```diff
@@ -971,6 +971,13 @@ uint8_t ull_cp_cte_req(struct ll_conn *conn, uint8_t min_cte_len, uint8_t cte_ty
 {
 	struct proc_ctx *ctx;
 
+	/* If controller gained,by any means, awareness of features supported by peer device
+	 * then check whether required features are enabled.
+	 */
+	if (!(conn->llcp.fex.features_used & BIT64(BT_LE_FEAT_BIT_CONN_CTE_RESP))) {
```
[the remark this answers is not in the document]
Comment (thoh-ot) [reply subsys/bluetooth/controller/ll_sw/ull_llcp.c:977]: Good catch; `llcp.fex.features_used` is initialized to the features of the local controller (`LL_FEAT`) and without checking `llcp.fex.valid` you cannot assume anything about the peer controller.

[the remark this answers is not in the document]
Comment (thoh-ot) [reply subsys/bluetooth/controller/ll_sw/ull_llcp.c:977]: Maybe also look into _ull_llcp_features.h_ and add any missing feature helper functions for CTE

**Comment by ppryga** [comment]
Fixed issue found by @cvinayak related with support for CTE RSP by peer device.

**Review by cvinayak — APPROVED** [review]

**Review by thoh-ot — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**